editors/intellij/src/main/kotlin/com/github/biomejs/intellijbiome/widgets/BiomeWidget.kt

```diff
@@ -1,16 +1,86 @@
+import com.github.biomejs.intellijbiome.listeners.BIOME_CONFIG_RESOLVED_TOPIC
+import com.github.biomejs.intellijbiome.listeners.BiomeConfigResolvedListener
+import com.github.biomejs.intellijbiome.lsp.BiomeLspServerSupportProvider
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
 import com.intellij.openapi.project.Project
-import com.intellij.openapi.wm.StatusBarWidget
+import com.intellij.openapi.wm.CustomStatusBarWidget
+import com.intellij.openapi.wm.StatusBarWidget.WidgetPresentation
+import com.intellij.openapi.wm.WindowManager
 import com.intellij.openapi.wm.impl.status.EditorBasedWidget
+import com.intellij.openapi.wm.impl.status.TextPanel.WithIconAndArrows
+import com.intellij.platform.lsp.api.LspServerManager
+import com.intellij.platform.lsp.impl.LspServerImpl
+import javax.swing.JComponent
 
-class BiomeWidget(myProject: Project) : EditorBasedWidget(myProject) {
+class BiomeWidget(project: Project) : EditorBasedWidget(project), CustomStatusBarWidget, WidgetPresentation {
+    private var component: WithIconAndArrows? = null
+
+    init {
+        project
+            .messageBus
+            .connect(this)
+            .subscribe(BIOME_CONFIG_RESOLVED_TOPIC, object : BiomeConfigResolvedListener {
+                override fun resolved(version: String) {
+                    update("Biome $version")
+                }
+            })
+    }
 
     override fun ID(): String {
         return "BiomeWidget"
     }
 
-    override fun getPresentation(): StatusBarWidget.WidgetPresentation {
-        return BiomeWidgetPresentation(project)
+    override fun getPresentation(): WidgetPresentation {
+        return this
+    }
+
+    override fun getComponent(): JComponent {
+        val component = WithIconAndArrows()
+        component.text = "Biome"
+        component.toolTipText = getTooltipText()
+
+        this.component = component
+
+        return component
+    }
+
+    override fun getTooltipText(): String {
+        val lspServerManager = LspServerManager.getInstance(project)
+        val lspServer = lspServerManager.getServersForProvider(BiomeLspServerSupportProvider::class.java).firstOrNull()
+
+        return when (lspServer) {
+            is LspServerImpl -> {
+                if (lspServer.isRunning) {
+                    BiomeBundle.message("biome.language.server.is.running")
+                } else {
+                    BiomeBundle.message("biome.language.server.is.stopped")
+                }
+            }
+
+            else -> {
+                BiomeBundle.message("biome.language.server.is.stopped")
+            }
+        }
     }
+
+    private fun update(text: String?) {
+        ApplicationManager.getApplication()
+            .invokeLater(
+                {
+                    if (project.isDisposed || component == null) {
+                        return@invokeLater
+                    }
+
+                    component!!.text = text
```

Comment:
Isn't there a risk that `component` is `null` here?

Reply:
The `null` check si together with `project.isDisposed` and it's an `OR` operation, this means that the project can be disposed but the component can still be null

Reply:
Done! made some code clean up